### ISSUE TITLE
Add FirebaseObject provider

### ISSUE THREAD

**Issue by jeffbcross** [angular/angularfire2#17]

**Comment by jeffbcross** [comment]
Any reason why it couldn't just be `FirebaseValue`, and support any kind of child value?

**Comment by davideast** [comment]
Unless we're keeping the DataSnapshot around (which isn't a terrible idea), we'll need to transform any primitive into an object to keep the key around.

Take the following data structure:

```json
{
  name: "Coco"
}
```

Then in AF2 we would do something like this:

```ts
class MyComponent {
  constructor(private af: AngularFire) {
    const obj = af.object('/name');
    obj.subscribe((data) => { 
      console.log(data);  // { value: "Coco", key: "name" }
    });
  }  
}
```

So we'll have to come up with a naming strategy for the `value` and `key` properties, but unless we return a DataSnapshot, we'll have to take this approach.

**Comment by jeffbcross** [comment]
Implemented in 35eba4e9610ebe19be7d4d14f00173d880e63cab
